test/e2e/storage/drivers/csi.go

```diff
@@ -110,54 +107,56 @@ func (h *hostpathCSIDriver) GetClaimSize() string {
 	return "5Gi"
 }
 
-func (h *hostpathCSIDriver) CreateDriver() {
+func (h *hostpathCSIDriver) CreateDriver(f *framework.Framework) (*testsuites.PerTestConfig, func()) {
 	By(fmt.Sprintf("deploying %s driver", h.driverInfo.Name))
-	f := h.driverInfo.Config.Framework
+	cancel := testsuites.StartPodLogs(f)
 	cs := f.ClientSet
 
 	// The hostpath CSI driver only works when everything runs on the same node.
 	nodes := framework.GetReadySchedulableNodesOrDie(cs)
 	nodeName := nodes.Items[rand.Intn(len(nodes.Items))].Name
-	h.driverInfo.Config.ClientNodeName = nodeName
+	config := &testsuites.PerTestConfig{
+		Driver:         h,
+		Prefix:         "hostpath",
+		Framework:      f,
+		ClientNodeName: nodeName,
+	}
 
 	// TODO (?): the storage.csi.image.version and storage.csi.image.registry
 	// settings are ignored for this test. We could patch the image definitions.
 	o := utils.PatchCSIOptions{
 		OldDriverName:            h.driverInfo.Name,
-		NewDriverName:            testsuites.GetUniqueDriverName(h),
+		NewDriverName:            config.GetUniqueDriverName(),
 		DriverContainerName:      "hostpath",
-		DriverContainerArguments: []string{"--drivername=csi-hostpath-" + f.UniqueName},
+		DriverContainerArguments: []string{"--drivername=" + config.GetUniqueDriverName()},
 		ProvisionerContainerName: "csi-provisioner",
 		NodeName:                 nodeName,
 	}
-	cleanup, err := h.driverInfo.Config.Framework.CreateFromManifests(func(item interface{}) error {
-		return utils.PatchCSIDeployment(h.driverInfo.Config.Framework, o, item)
+	cleanup, err := config.Framework.CreateFromManifests(func(item interface{}) error {
+		return utils.PatchCSIDeployment(config.Framework, o, item)
 	},
 		h.manifests...)
-	h.cleanup = cleanup
 	if err != nil {
 		framework.Failf("deploying %s driver: %v", h.driverInfo.Name, err)
 	}
-}
 
-func (h *hostpathCSIDriver) CleanupDriver() {
-	if h.cleanup != nil {
+	return config, func() {
 		By(fmt.Sprintf("uninstalling %s driver", h.driverInfo.Name))
-		h.cleanup()
+		cleanup()
+		cancel()
```

Comment:
should we be cancelling pod logging before or after cleanup? Also more descriptive names please. `cleanupDriver` and `cancelLogging`?

Reply:
okay